src/App.js

```diff
@@ -1,26 +1,198 @@
+          </div>
+        </span>
+      </div>
+    )
+  }
+}
+
+
+//////Iteration2//////
+function LangFr(props) {
+  return(
+  <div style={{width: "800px", height:"80px", margin: "10px", border: "1px solid black" }}>
+  <h1>Bonjour François</h1>
+  </div>
+  )
+}
+
+function LangDe(props) {
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div style={{width: "800px", height:"80px", margin: "10px", border: "1px solid black" }}>
+    <h1>Hallo Ludwig</h1>
     </div>
+  )
+}
+
+function Greetings(props) {
+    const language = props.lang;
+    if (language === "fr" ) {
+      return <LangFr />
+    }
+    return <LangDe /> 
+  }
+
+//////Iteration3//////
+
+
+function Random (props) {
+    const min = props.min;
+    const max = props.max;
+    const rand = Math.trunc (Math.random() * (max - min));
+
+    return(
+    <div style={{width: "800px", height:"80px", margin: "10px", border: "1px solid black" }}>
+    <h1>Random value between {props.min} and {props.max} => {rand}</h1>
+    </div>
+    )
+}
+
+//////Iteration4//////
+
+function BoxColor (props) {
+
+  function componentToHex(c) {
+    var hex = c.toString(16);
+    return hex.length === 1 ? "0" + hex : hex;
+  }
+  const hex = ("#" + componentToHex(props.r) + componentToHex(props.g) + componentToHex(props.b))
+  
+  return(
+    <div style={{backgroundColor: `${hex}`, width: "800px", height:"150px", margin: "10px", border: "1px solid black" }}>
+    <h3 style={{padding: "30px 40%"}}>rgb({props.r},{props.g},{props.b}) {hex}</h3>
+    </div>
+  )
+}
+
+//////Iteration5//////
+
+
+function CreditCard (props) {
+  const cardTypeToLogo = {
+    'Master Card': masterCardLogo,
+    'Visa': visaLogo};
+
+    function fourNum (num) {
+      return "xxxx xxxx xxxx " + num.substr(num.length-4)
+    }
+
+    
+  return(
+    <div className='credit-card' style={{backgroundColor: [props.bgColor]}}>
+    <div className='credit-card__logo'>
+        <img className='cardlogo' src={cardTypeToLogo[props.type]} width="30px"/>
+    </div>
+
+    <div className='credit-card__number'>{fourNum(props.number)}</div>
+    
+    <div className='credit-card__info'>
+        <div className='credit-card__info_name'>
+            <div className='credit-card__info_label'>{props.owner}</div>
+            <div>{props.bank}</div>
+        </div>
+
+        <div className='credit-card__info_expiry'>
+            <div className='credit-card__info_label'>VALID UP TO</div>
+            <div>{props.expirationMonth}/{props.expirationYear}</div>
+        </div>
+    </div>
+    </div>
+  )
+}
+
+/////////////////////
+
+function App() {
```

Comment:
Overall brother you did a good job! Just creat components and organize everything and use the switch statement in the second iteration to not repeat yourself!